docs/advanced_features.md

```diff
@@ -0,0 +1,71 @@
+# Advanced features
+
+## SPARQL+Text and SPARQL Autocompletion
+
+On top of the vanilla SPARQL functionality, QLever allows so-called SPARQL+Text
+queries on a text corpus linked to a knowledge base via entity recognition.  For
+example, the following query find all mentions of astronauts next to the words
+"moon" and "walk*" in the text corpus:
+
+    SELECT ?a TEXT(?t) SCORE(?t) WHERE {
+        ?a <is-a> <Astronaut> .
+        ?t ql:contains-entity ?a .
+        ?t ql:contains-word "walk* moon"
+    } ORDER BY DESC(SCORE(?t))
+
+Such queries can be simulated in standard SPARQL, but only with poor
+performance, see the CIKM'17 paper above.  Details about the required input data
+and the SPARQL+text query syntax and semantics can be found
+[here](docs/sparql_plus_text.md).
+
+QLever also supports efficient SPARQL autocompletion.  For example, the
+following query yields a list of all predicates associated with persons in the
+knowledge base, ordered by the number of persons which have that predicate.
+
+    SELECT ?predicate (COUNT(?predicate) as ?count) WHERE {
+      ?x <is-a> <Person> .
+      ?x ql:has-predicate ?predicate
+    }
+    GROUP BY ?predicate
+    ORDER BY DESC(?count)
+
+Note that this query could also be processed by standard SPARQL simply by
+replacing the second triple by ?x ?predicate ?object. However, that query is
+bound to produce a very large intermediate result (all triples of all persons)
+with a correspondingly huge query time.  In contrast, the query above takes only
+~ 100ms on a standard Linux machine for a dataset with ~ 360 million triples and
```

Comment:
This is down to personal preference, but I would write `about` instead of using `~` to interrupt the text flow less. Also, some rough information on the hardware of the linux machine could be interesting here, as I am otherwise uncertain what level of standard you refer to (standard for a consumer pc, or a compute server, ...)

Reply:
Hmm, the latter really depends on the KB. The example and the original text (written by Björn) is about Freebase Easy, which nowadays works well with 16 GB RAM so pretty much a beefy laptop. On the other hand we now think of Wikidata as the standard data set and that still needs 64 GB

Reply:
Since it says 360 million triples I think we should just mention 16 GB and it still works for "standard workstation"